package/MDAnalysis/wham.py

```diff
@@ -0,0 +1,644 @@
+
+    # check our reaction coord type option...
+    calc_reaction_coord(timeseries_type)
+
+    # check start/end time values...
+    _check_min_max(start_time, "start time", end_time, "end_time")
+
+    # check our energy unit options...
+    convert_energy(unitin=energy_units_in, unitout=energy_units_out)
+    energy_units_wham = 'kcal'
+    ## TODO - we still need to know if wham is configured for kcal or kj - 
+    ## check this from the file, or require as user input?
+
+    # set up files...
+    try:
+        os.makedirs(root)
+    except OSError:
+        if not keep_files:
+            raise ValueError("Folder {} already exists".format(root))
+        # otherwise assume it's fine to overwrite stuff?
+    # TODO - also allow different names for these...
+    meta_fname = root+'/metadatafile.dat'
+    data_fname = './timeseries_{}.dat' # assuming they're in the same dir as meta
+    out_fname = root+'/output.dat'
+
+    # check all our wham parameters are valid before we start writing files
+    run_grossfield_wham(check_only=True, calc_temperature=calc_temperature,
+                        hist_max=hist_max, hist_min=hist_min, 
+                        outfile=out_fname, metafile=meta_fname, **wham_args)
+
+    # write input files
+    min_max = write_grossfield_input(bundle, data_names=data_names, 
+                                     filename=meta_fname, data_fname=data_fname,
+                                     hist_max=hist_max, hist_min=hist_min, 
+                                     start_time=start_time, end_time=end_time,
+                                     timeseries_type=timeseries_type, 
+                                     ener_units=energy_units_in, 
+                                     wham_units=energy_units_wham,
+                                     checked=True)
+
+    # set our hist min/max values if not specified in args
+    hist_min = min_max[0] if hist_min is None else hist_min
+    hist_max = min_max[1] if hist_max is None else hist_max
+
+    # run!
+    run_grossfield_wham(calc_temperature=calc_temperature, hist_max=hist_max, 
+                        hist_min=hist_min, metafile=meta_fname,
+                        outfile=out_fname, **wham_args)
+    # read the output    
+    profile = read_grossfield_output(out_fname, ener_units=energy_units_out,
+                                     wham_units=energy_units_wham,
+                                     temperature=calc_temperature)
+
+    # clear our files
+    if not keep_files:
+        # should only have been created by us, so should be good to remove...
+        shutil.rmtree(root)
+
+    return profile
+    # TODO - in the outfile we also get the probability + it's error in [:,3] 
+    # and [:,4]; and the 'F-values' for each simulation (that we didn't skip);
+    # option to get prob instead of PMF? option to return Fvalues as well (tuple?)
+
+
+def check_names(bundle, spring='spring', loc_win_min='loc_win_min', 
```

Comment:
Couldn't this function be more generic and take a list of names as argument? Maybe something in the line of:

```python
def check_names(bundle, mandatories, optionals)
```